pkg/eks/api.go

```diff
@@ -377,6 +377,13 @@ func (c *ClusterProvider) SetAvailabilityZones(spec *api.ClusterConfig, given []
 		return nil
 	}
 
+	if count := len(spec.AvailabilityZones); count != 0 {
+		if count < api.MinRequiredAvailabilityZones {
+			return api.ErrTooFewAvailabilityZones(spec.AvailabilityZones)
+		}
+		return nil
+	}
+
```

Comment:
don't we already do this? https://github.com/weaveworks/eksctl/blob/df75054cf0b2696130f465ac7a0af7e4bbe7b851/pkg/apis/eksctl.io/v1alpha5/validation.go#L1296

Reply:
For given azs, but those don't come from the spec apparently.

Reply:
They come from `params`:

```go
		if err := ctl.SetAvailabilityZones(cfg, params.AvailabilityZones); err != nil {
			return err
		}
```

Reply:
Which is set by cli. But the extra check is for when it's coming from the yaml config. Pretty weird, could probably be refactored at some point.

Reply:
> For given azs, but those don't come from the spec apparently.

I don't follow, both codes use `spec/cfg.AvailabilityZones`

Reply:
Very good point but no it doesn't: this specific function is (only) used by `doCreateCluster`, which does not use `ValidateClusterConfig`. Moved this validation to `ValidateClusterConfig`, but the next step is to make all these `ctl` funcs actually use it. https://github.com/weaveworks/eksctl/issues/4862 addresses this :)

Reply:
See previous comment. :)